pkg/controller/admissionchecks/multikueue/workload.go

```diff
@@ -442,6 +445,7 @@ func newWlReconciler(c client.Client, helper *multiKueueStoreHelper, cRec *clust
 		deletedWlCache:    utilmaps.NewSyncMap[string, *kueue.Workload](0),
 		eventsBatchPeriod: eventsBatchPeriod,
 		adapters:          adapters,
+		clock:             realClock,
```

Comment:
We can't use `FakeClock` this way, so this solution has no benefit—`realClock.Now()`and `time.Now()` are essentially the same.

Since we aren’t using Option, maybe we should just pass it directly to the constructor?

Reply:
You mean something like func newWlReconciler(c client.Client, helper *multiKueueStoreHelper, cRec *clustersReconciler, origin string, workerLostTimeout, eventsBatchPeriod time.Duration, adapters map[string]jobframework.MultiKueueAdapter, clk clock.Clock) *wlReconciler ??

Reply:
Yeah, something like that I think.

Reply:
Does it work?

Reply:
On the other hand we also can implement Options here.

Reply:
I think with Options solution is better but takes more changes.

Reply:
If it's better then i have no issue in implementing that. I think `pkg/scheduler` has it. I just have to do the same thing here.